Mathlib/Tactic/Congr!.lean

```diff
@@ -0,0 +1,248 @@
+/-
+Copyright (c) 2023 Kyle Miller. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kyle Miller
+-/
+import Lean
+import Mathlib.Tactic.Relation.Rfl
+
+/-!
+# The `congr!` tactic
+
+This is a more powerful version of the `congr` tactic.
```

Comment:
Can you explain (somewhere) how this relates to Lean 3 `congr`/`congr'`/`convert`?

Reply:
I added a little bit to relate them.